### PR TITLE
Feature/sdk bump

### PR DESCRIPTION
**Pull request by bowtiedpyro**
Merge sdk_bump into release/redesign - latest version of SDK, updated functions and structure. UI updates, market data update

### CODE REVIEW

File: contexts/AccountsProvider.tsx [hunk 1]
```diff
@@ -124,25 +124,42 @@ export const AccountsProvider: React.FC<React.ReactNode> = ({ children }) => {
...
+const allowedCollections = [
+  'pesky penguins',
+  'burrito boyz',
+  'droid',
+  'lifinity flares',
+  'vandals',
+  'honey genesis bee',
+  'elixir: ovols',
+  'blocksmith labs',
+  "Trippin' Ape Tribe"
```
Comment (BowTiedHeron) [inline contexts/AccountsProvider.tsx:136]: need to be expanded or deprecated.

File: helpers/loanHelpers/userCollection.tsx [hunk 2]
```diff
@@ -56,130 +72,15 @@ import {
...
-          .div(new BN(10 ** 15))
-          .toNumber() *
-          1.002) /
-        LAMPORTS_PER_SOL;
-    }
-  }
-  return RoundHalfDown(nftCollateralValue * MAX_LTV - userLoans, 4);
-}
-
 export async function fetchLTV(totalMarketDebt: number, nftPrice: number) {
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:75]: still needs to be exported? @bowtiedpyro

File: helpers/loanHelpers/userCollection.tsx [hunk 3]
```diff
@@ -56,130 +72,15 @@ import {
...
-  }
-
-  return RoundHalfUp(totalDebt);
-}
 // filters out zero debt obligations and multiplies outstanding obl. by nft price
-export async function fetchTVL(nftPrice: number, obligations: any) {
-  return nftPrice * obligations.filter((obl: any) => obl.debt !== 0).length;
+export async function fetchTVL(obligations: any) {
+  if (!obligations.length) return 0;
+  return obligations.filter((obl: any) => obl.debt !== 0).length;
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:83]: also needs to be exported?

File: helpers/loanHelpers/userCollection.tsx [hunk 4]
```diff
@@ -213,6 +114,22 @@ export async function calcNFT(
     return 0;
   }
 }
+
+/**
+ * @description pollutes the chart on lend with dummy historic rates
+ * @params none
+ * @returns chart data
+ */
+const getPositionData = () => {
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:123]: is this relevant? dummy chart data?

File: helpers/loanHelpers/userCollection.tsx [hunk 5]
```diff
@@ -342,7 +176,7 @@ const setObligations = async (
         highestBid: obligation.highest_bid
       };
     })
-    .filter((obl: any) => obl.debt !== 0);
+    .filter((obl: any) => obl.debt.toString() != 0);
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:179]: @bowtiedpyro verify if this function can be removed (as discussed)

File: helpers/loanHelpers/userCollection.tsx [hunk 6]
```diff
@@ -356,80 +190,135 @@ const calculateRisk = async (
...
   let sumOfDebt = await filtered.reduce((acc: number, obligation: any) => {
-    return acc + obligation.debt;
+    return (acc += obligation.debt);
   }, 0);
 
   if (type === true) {
     return sumOfDebt;
   } else {
-    return sumOfDebt / filtered.length / nftPrice;
+    const sum = (sumOfDebt / filtered.length / nftPrice) * 100;
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:204]: was this *100 migrated from sdk to frontend?
verify if it wasnt correct at first/if its correct now

File: helpers/loanHelpers/userCollection.tsx [hunk 7]
```diff
@@ -440,98 +329,50 @@ async function handleFormatMarket(
...
-    collection.value = totalMarketValue;
-    collection.connection = connection;
-    collection.utilizationRate = Number(
-      f(totalMarketDebt / (totalMarketDeposits + totalMarketDebt))
-    );
-    collection.user = honeyUser;
-    collection.nftPrice = nftPrice;
-    collection.ltv = ltv;
-    collection.tvl = tvl;
+  const tvl = new BN(nftPrice * (await fetchTVL(obligations)));
```
Comment (BowTiedHeron) [inline helpers/loanHelpers/userCollection.tsx:356]: move BN to sdk?

File: helpers/marketHelpers/index.tsx [hunk 8]
```diff
@@ -333,6 +288,102 @@ export const marketCollections: MarketTableRow[] = [
       discountedMarket: false
     }
   }
+  // {
+  //   id: '5UKRRSxbi4PgPnQU2ZqtukUxd1fyN6ydn1hoxivP46A8',
+  //   key: 'OLV',
```
Comment (BowTiedHeron) [inline helpers/marketHelpers/index.tsx:293]: double

**Review by BowTiedHeron — CHANGES_REQUESTED** [review]
need to finish reviewing borrow, lend and liquidations tomorrow morning, as well as walking through comments.

File: pages/borrow/index.tsx [hunk 9]
```diff
@@ -196,17 +201,27 @@ const Markets: NextPage = () => {
...
-
   const [isCreateMarketAreaOnHover, setIsCreateMarketAreaOnHover] =
     useState<boolean>(false);
 
   // fetches the sol price
   // TODO: create type for reserves and connection
-  async function fetchSolValue(reserves: any, connection: any) {
-    const slPrice = await fetchSolPrice(reserves, connection);
+  async function fetchSolValue(reserves: TReserve, connection: Connection) {
+    const slPrice = await fetchReservePrice(reserves, connection);
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:210]: rename constant to reserve price or something related to SPL tokens.
first it was SOL, now not anymore.

File: pages/borrow/index.tsx [hunk 10]
```diff
@@ -196,17 +201,27 @@ const Markets: NextPage = () => {
...
   }
 
+  /**
+   * @description sets state of marketValue by parsing lamports outstanding debt amount to SOL
+   * @params none, requires parsedReserves
+   * @returns updates marketValue
+   */
+  useEffect(() => {
+    if (parsedReserves) {
+      fetchSolValue(parsedReserves[0], sdkConfig.saberHqConnection);
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:221]: rename solValue

File: pages/borrow/index.tsx [hunk 11]
```diff
@@ -217,50 +232,23 @@ const Markets: NextPage = () => {
...
-    }
-  }, [parsedReserves]);
-
-  // sets cRatio, liquidationThreshold and calls fetchHelperValues
-  useEffect(() => {
-    if (marketReserveInfo && parsedReserves) {
-      setCRatio(BnToDecimal(marketReserveInfo[0].minCollateralRatio, 15, 5));
-    }
-  }, [marketReserveInfo, parsedReserves]);
+    const marketIDs = marketCollections.map(market => market.id);
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:239]: @bowtiedpyro create marketID array based on loop through market collections.
don't want to keep track of markets at multiple places, but reuse values.

File: pages/borrow/index.tsx [hunk 12]
```diff
@@ -285,10 +278,12 @@ const Markets: NextPage = () => {
                 marketObject =>
                   marketObject.market.address.toString() === collection.id
               );
+
               const honeyUser = collection.marketData[0].user;
               const honeyMarket = collection.marketData[0].market;
               const honeyClient = collection.marketData[0].client;
               const parsedReserves = collection.marketData[0].reserves[0].data;
+              const mData = collection.marketData[0].reserves[0];
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:286]: consider renaming mData to something more self explanatory/obvious (keeping 3rd parties in mind)

File: pages/borrow/index.tsx [hunk 13]
```diff
@@ -302,74 +297,44 @@ const Markets: NextPage = () => {
...
-                (await getInterestRate(
-                  collection.utilizationRate,
-                  collection.id
-                )) || 0;
-
               if (currentMarketId === collection.id) {
+                setActiveInterestRate(collection.rate);
                 // @ts-ignore
-                setNftPrice(RoundHalfDown(collection.nftPrice));
+                setNftPrice(RoundHalfDown(Number(collection.nftPrice)));
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:312]: still BN or string? is conversion to Number required?

File: pages/borrow/index.tsx [hunk 14]
```diff
@@ -1017,54 +1018,66 @@ const Markets: NextPage = () => {
   async function executeRepay(val: any, toast: ToastProps['toast']) {
     try {
       if (!val) return toast.error('Please provide a value');
+      // add additional value if user wants to repay 100% of loan due to interest rate not being included
+      if (val == userDebt) val += 1;
```
Comment (BowTiedHeron) [inline pages/borrow/index.tsx:1022]: @bowtiedpyro update to value used in SSR branch as discussed
make sure that all related updates regarding the SDK bump are included in this PR/branch 
(no additional ones in the SSR branch)

File: pages/lend/index.tsx [hunk 15]
```diff
@@ -171,49 +168,35 @@ const Lend: NextPage = () => {
...
-  //   if (marketReserveInfo && honeyUser)
-  //     calculateTotalUserDeposits(marketReserveInfo, honeyUser);
-  // });
-  //  ************* END CALC. USER DEPOSITS *************
-
   //  ************* START FETCH CURRENT SOL PRICE *************
   // fetches the current sol price
-  async function fetchSolValue(reserves: any, connection: any) {
-    const slPrice = await fetchSolPrice(reserves, connection);
+  async function fetchSolValue(reserves: TReserve, connection: Connection) {
```
Comment (BowTiedHeron) [inline pages/lend/index.tsx:173]: rename func as mentioned for borrow page

File: pages/lend/index.tsx [hunk 16]
```diff
@@ -171,49 +168,35 @@ const Lend: NextPage = () => {
...
+    if (parsedReserves) {
+      fetchSolValue(parsedReserves[0], sdkConfig.saberHqConnection);
+    }
+  }, [parsedReserves]);
   //  ************* END FETCH CURRENT SOL PRICE *************
 
   /**
-   * @description deposits 1 sol
-   * @params optional value from user input; amount of SOL
+   * @description deposits X amount of SOL from market
```
Comment (BowTiedHeron) [inline pages/lend/index.tsx:191]: only sol?

File: pages/lend/index.tsx [hunk 17]
```diff
@@ -237,41 +220,51 @@ const Lend: NextPage = () => {
...
       }
     } catch (error) {
-      return toast.error('Deposit failed', error);
+      return toast.error('Deposit failed');
     }
   }
   /**
-   * @description withdraws 1 sol
-   * @params optional value from user input; amount of SOL
+   * @description withdraws X amount of SOL from market
```
Comment (BowTiedHeron) [inline pages/lend/index.tsx:258]: only sol?

File: pages/lend/index.tsx [hunk 18]
```diff
@@ -378,60 +382,38 @@ const Lend: NextPage = () => {
...
-                  collection.id
-                )) || 0) * collection.utilizationRate;
-
-              collection.stats = getPositionData();
-
-              if (currentMarketId == collection.id) {
-                setActiveMarketSupplied(collection.value);
-                setActiveMarketAvailable(collection.available);
-              }
+              setTimeout(() => {
```
Comment (BowTiedHeron) [inline pages/lend/index.tsx:401]: ideally we dont use fixed timeouts, but based on loading state

File: pages/liquidate/index.tsx [hunk 19]
```diff
@@ -252,41 +249,8 @@ const Liquidate: NextPage = () => {
...
-
-  //  ************* START FETCH SOL PRICE *************
-  /**
-   * @description
-   * @params
-   * @returns
-   */
-  async function fetchSolValue(reserves: any, connection: any) {
-    const slPrice = await fetchSolPrice(reserves, connection);
+  async function fetchSolValue(reserves: TReserve, connection: Connection) {
```
Comment (BowTiedHeron) [inline pages/liquidate/index.tsx:252]: rename solValue

File: pages/liquidate/index.tsx [hunk 20]
```diff
@@ -333,10 +296,10 @@ const Liquidate: NextPage = () => {
 
           if (transactionOutcome[0] == 'SUCCESS') {
             setCurrentUserBid(0);
-
+            if (walletPK) await fetchWalletBalance(walletPK);
```
Comment (BowTiedHeron) [inline pages/liquidate/index.tsx:299]: add wait for confirmation? @bowtiedpyro

File: pages/liquidate/index.tsx [hunk 21]
```diff
@@ -353,14 +316,15 @@ const Liquidate: NextPage = () => {
           });
 
           if (transactionOutcome[0] == 'SUCCESS') {
+            if (walletPK) await fetchWalletBalance(walletPK);
             return toast.success('Bid placed, fetching chain data');
           } else {
-            return toast.error('Bid failed');
+            return toast.error('Bid failed', transactionOutcome);
```
Comment (BowTiedHeron) [inline pages/liquidate/index.tsx:322]: remove tx outcome

File: pages/liquidate/index.tsx [hunk 22]
```diff
@@ -333,10 +296,10 @@ const Liquidate: NextPage = () => {
 
           if (transactionOutcome[0] == 'SUCCESS') {
             setCurrentUserBid(0);
-
+            if (walletPK) await fetchWalletBalance(walletPK);
             return toast.success('Bid revoked, fetching chain data');
           } else {
-            return toast.error('Revoke bid failed');
+            return toast.error('Revoke bid failed', transactionOutcome[0]);
```
Comment (BowTiedHeron) [inline pages/liquidate/index.tsx:302]: remove tx outcome

File: pages/liquidate/index.tsx [hunk 23]
```diff
@@ -353,14 +316,15 @@ const Liquidate: NextPage = () => {
...
             return toast.success('Bid placed, fetching chain data');
           } else {
-            return toast.error('Bid failed');
+            return toast.error('Bid failed', transactionOutcome);
           }
         } else if (type == 'increase_bid') {
           // if no user bid terminate action
           if (!userBid) return;
-
+          userBid = Number(userBid.toFixed(2));
```
Comment (BowTiedHeron) [inline pages/liquidate/index.tsx:327]: @bowtiedpyro sync with @yuriagile if this should be Number or BN

**Review by BowTiedHeron — CHANGES_REQUESTED** [review]
added final comments that need to be reviewed before merge
will sync again before merging!

File: contexts/AccountsProvider.tsx (hunk 1 above)
Comment (bowtiedpyro) [reply contexts/AccountsProvider.tsx:136]: Done.

File: helpers/loanHelpers/userCollection.tsx (hunk 2 above)
Comment (bowtiedpyro) [reply helpers/loanHelpers/userCollection.tsx:75]: For now yes, will check with sdk / open request for shipping it there.

File: helpers/loanHelpers/userCollection.tsx (hunk 4 above)
Comment (bowtiedpyro) [reply helpers/loanHelpers/userCollection.tsx:123]: Is required for future work - we should leave it there for now.

File: helpers/loanHelpers/userCollection.tsx (hunk 5 above)
Comment (bowtiedpyro) [reply helpers/loanHelpers/userCollection.tsx:179]: This formats our obligations as presented in the FED, for now we need this. Will bring it up with SDK dev to validate if we want to ship this logic to the SDK

File: helpers/loanHelpers/userCollection.tsx (hunk 7 above)
Comment (bowtiedpyro) [reply helpers/loanHelpers/userCollection.tsx:356]: doesn't need to be BN - converted it back, simply is a calculation now - but will discuss this to ship to SDK

File: helpers/loanHelpers/userCollection.tsx (hunk 6 above)
Comment (bowtiedpyro) [reply helpers/loanHelpers/userCollection.tsx:204]: it's the forumla for the risk factor, its the total sum of debt - div. by the length of the array div. by the nft price, which gives us a 00,XX% value - so we multiply by 100 to get a XX,XX% value

File: pages/borrow/index.tsx (hunk 11 above)
Comment (bowtiedpyro) [reply pages/borrow/index.tsx:239]: fixed, currently not in use - works: future work we can implement this.

File: helpers/marketHelpers/index.tsx (hunk 8 above)
Comment (bowtiedpyro) [reply helpers/marketHelpers/index.tsx:293]: removed

File: pages/borrow/index.tsx (hunk 9 above)
Comment (bowtiedpyro) [reply pages/borrow/index.tsx:210]: done

File: pages/borrow/index.tsx (hunk 12 above)
Comment (bowtiedpyro) [reply pages/borrow/index.tsx:286]: nice to have

File: pages/borrow/index.tsx (hunk 13 above)
Comment (bowtiedpyro) [reply pages/borrow/index.tsx:312]: fixed, bn was not needed, nor number

File: pages/borrow/index.tsx (hunk 14 above)
Comment (bowtiedpyro) [reply pages/borrow/index.tsx:1022]: done

File: pages/lend/index.tsx (hunk 15 above)
Comment (bowtiedpyro) [reply pages/lend/index.tsx:173]: done

File: pages/lend/index.tsx (hunk 17 above)
Comment (bowtiedpyro) [reply pages/lend/index.tsx:258]: done

File: pages/lend/index.tsx (hunk 16 above)
Comment (bowtiedpyro) [reply pages/lend/index.tsx:191]: done

File: pages/lend/index.tsx (hunk 18 above)
Comment (bowtiedpyro) [reply pages/lend/index.tsx:401]: true, for now we leave it like this - need to see where the delay is

File: pages/liquidate/index.tsx (hunk 19 above)
Comment (bowtiedpyro) [reply pages/liquidate/index.tsx:252]: done

File: pages/liquidate/index.tsx (hunk 21 above)
Comment (bowtiedpyro) [reply pages/liquidate/index.tsx:322]: done

File: pages/liquidate/index.tsx (hunk 22 above)
Comment (bowtiedpyro) [reply pages/liquidate/index.tsx:302]: done

File: pages/liquidate/index.tsx (hunk 23 above)
Comment (bowtiedpyro) [reply pages/liquidate/index.tsx:327]: will need to validate with Yuri, for now we can leave it like this

File: pages/liquidate/index.tsx (hunk 20 above)
Comment (bowtiedpyro) [reply pages/liquidate/index.tsx:299]: Will test this out

**Pull request closed without merging**